relay-general/src/store/normalize.rs

```diff
@@ -264,18 +264,45 @@ fn remove_invalid_measurements(
     measurements: &mut Measurements,
     meta: &mut Meta,
     measurements_config: &MeasurementsConfig,
+    max_name_and_unit_len: Option<isize>,
 ) {
     let mut custom_measurements_count = 0;
     let mut removed_measurements = Object::new();
 
     measurements.retain(|name, value| {
-        let measurement = match value.value() {
+        let measurement = match value.value_mut() {
             Some(m) => m,
             None => return false,
         };
+
+        if !is_valid_metric_name(name) {
+            meta.add_error(Error::invalid(format!(
```

Comment:
that sounds more right yeah, i'll change it. I used iinvalid because that was used in the already present error message but i see that it might not fit here as much

Reply:
hmm although it seems you lose some information then, like you can put InvalidData and write the original value, but you don't get the exact reason why it's invalid?